FF1Lib/Armor.cs

```diff
@@ -81,6 +92,362 @@ public void ExpandArmor()
 				EquipPermission.WhiteWizard));
 			platinumBracelet.writeArmorMemory(this);
 		}
+
+		public void ArmorCrafter(MT19337 rng) {
+		    var tierList = new List<IReadOnlyList<Item>> { ItemLists.CommonArmorTier,
+								   ItemLists.RareArmorTier,
+								   ItemLists.LegendaryArmorTier };
+
+		    // armor classes, determines who can equip
+		    // cloth -- everyone
+		    // light -- fi/th/bb/rm
+		    // medium -- fi/ni/rm
+		    // heavy -- fi/ni
+		    // legendary (knight, mage stuff, ribbon, 1 more?)
+
+		    const int CLOTH = 0;
+		    const int LIGHT = 1;
+		    const int MEDIUM = 2;
+		    const int HEAVY = 3;
+		    const int KNIGHT = 4;
+
+		    // Armor types.
+		    const int ARMOR = 0;
+		    const int SHIELD = 1;
+		    const int HELM = 2;
+		    const int GAUNTLET = 3;
+		    const int BRACELET = 4;
+		    const int SHIRT = 5;
+		    const int CAPE = 6;
+		    const int RING = 7;
+
+		    // class x type -> defense base
+		    var armorDefenseBase = new int[,] {
+			// armor, shield, helm, gauntlet, bracelet, shirt, cape, ring
+			{  4,       2,    1,        1,        4,     6,    8,    8 },  // cloth
+			{ 10,       4,    3,        2,        8,    12,    8,    8 },  // light
+			{ 20,       8,    6,        4,       16,    16,    8,    8 },  // medium
+			{ 30,      12,    6,        6,       24,    24,    8,    8 },  // heavy
+			{ 42,      16,    8,        8,       32,    24,    8,    8 },  // knight
+		    };
+

```

Comment:
I get from below, that there's only 5 knight gear items(the opal set). The others are created from predefined items(white shirt...) a required list and random. From what I understand, it can create cloth/light/medium rings and capes, which are exactly the same but not equippable by everyone.

Reply:
There are 4 pieces in the knight set, not 5.

"CLOTH" is a quality level (armor class), any armor type can be "CLOTH" quality, it means it rolls low defense but is equippable by anyone.

Capes are equippable by anyone except black belt/master (like in vanilla), and occupy the shield slot.  They can be any quality level up to HEAVY.

Rings are equippable by everyone, and occupy the gauntlet slot.  They can be any quality level up to HEAVY.

Reply:
The armorClass doesn't seem to make a difference for rings and capes. Is it warranted to have theese different classes? I mean since every class/type combination has the same chance of appearing(excluding your requiredList) there will be more sameish rings and capes.
Is that intentional?

Reply:
Yes you're right, that's not ideal.  I adjusted the table.